mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulSpeechRecognitionScreenTest.kt

```diff
@@ -0,0 +1,57 @@
+package ch.epfl.sdp.mobile.test.state
+
+import android.Manifest
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithContentDescription
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.rule.GrantPermissionRule.grant
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.authentication.AuthenticationFacade
+import ch.epfl.sdp.mobile.application.chess.ChessFacade
+import ch.epfl.sdp.mobile.application.social.SocialFacade
+import ch.epfl.sdp.mobile.state.ProvideFacades
+import ch.epfl.sdp.mobile.state.StatefulSpeechRecognitionScreen
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.auth.emptyAuth
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.store.emptyStore
+import ch.epfl.sdp.mobile.ui.speech_recognition.DefaultText
+import ch.epfl.sdp.mobile.ui.speech_recognition.ListeningText
+import ch.epfl.sdp.mobile.ui.speech_recognition.MicroIconDescription
+import ch.epfl.sdp.mobile.ui.speech_recognition.PermissionGranted
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalPermissionsApi
+class StatefulSpeechRecognitionScreenTest {
+
+  @get:Rule val rule = createComposeRule()
+
+  @get:Rule val permissionRule: GrantPermissionRule = grant(Manifest.permission.RECORD_AUDIO)
+
+  @Test
+  fun given_defaultScreen_when_micClicked_and_okClicked_then_permissionGranted() = runTest {
```

Comment:
I think this would be more in line with our new naming conventions.
```suggestion
  fun given_defaultScreen_when_micClickedAndOkClicked_then_permissionGranted() = runTest {
```

Reply:
Thanks, actually, the whole name needs to be refactored since its content does not correspond anymore to its name.